.github/check-for-changes.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+if [ $(git diff --no-ext-diff --quiet --exit-code) ]; then
+    exit 0
+else
+    git log
+    git diff --name-only
+    exit 1
+fi
```

Comment:
I think this whole thing could just be

```shell
git diff --exit-code
```

?

In which case, we might as well inline it into the CI config, rather than put it in a script.

Not sure what the `git log` is buying us, but that is going to log a lot of things. Maybe it is supposed to have `-1`? Either way, I don't think it is too useful because surely you know what the HEAD is when looking at CI.

Reply:
Um, yeah, that is me trying to figure out what the CI is doing 😄.

Reply:
Yeah, I think now that I understand this a bit better I am going to remove the script and go with `git diff --no-ext-diff --name-only --exit-code`:
 - `--no-ext-diff` just in case there's a weird diff tool setup (highly unlikely)
 - `--name-only` to print the file names (but not the bytes) so we know which files changed
 - `--exit-code` to pass/fail the build.